### ISSUE TITLE
Unable to insert into table with identity primary key using Hasura Console

### ISSUE THREAD

**Issue by devuxer** [hasura/graphql-engine#4863]
Possibly related to #3572.

Using v1.2.1 of console and server, if I try to insert a row into a table with an auto-incrementing primary key using Hasura Console > Data > Insert Row, I get this error:

> Insert failed!
> unexpected : cannot insert into column "id"

My id column is defined as `integer, primary key, unique`.

There doesn't seem to be a way to leave the id out of the insert, and I'm not able to select the "null" or "default" radio button. Also, if I enter an ID that's not already in the table, it still doesn't work.

**Comment by rikinsk** [comment]
@devuxer Can you confirm is your primary key is indeed auto-incrementing? If so, you should be able to see a default value such as `nextval('author_id_seq'::regclass)` for it as shown here:
![Screenshot from 2020-05-29 16-50-58](https://user-images.githubusercontent.com/54616969/83254589-aad97600-a1cc-11ea-9d69-101bd4097822.png).

In case the id is a generated column then this is a known issue as you have mentioned and should be fixed in the next release.

**Comment by devuxer** [comment]
@rikinsk,

I created the column in the first place using the following

```sql
-- create table
create table "Member" (
    "id" integer primary key generated by default as identity,
    -- [other columns...]
);

-- insert seed data that already has IDs

-- once seed data is inserted, modify id column to be **always** auto-incrementing
alter table "Member" alter column id set generated always;

-- set the current ID to the max ID so far
select setval( pg_get_serial_sequence('"Member"', 'id'), (select max(id) from "Member"));
```

The above process apparently does not cause `nextval` to appear in the column definition in Hasura. I only see `integer, primary key, unique`.

![image](https://user-images.githubusercontent.com/1298521/83289675-3c7bc000-a19a-11ea-9f8e-3afb1b497de8.png)

However, if I insert a row using pgAdmin without specifying an ID, the ID does auto-increment as expected, and the schema indicates `identity_generation = ALWAYS` and `identity_increment = 1`:

![image](https://user-images.githubusercontent.com/1298521/83289938-aa27ec00-a19a-11ea-99fd-05bc487ac12c.png)

Hopefully, that's enough detail, but let me know if you need anything else.

**Comment by rikinsk** [comment]
I think the issue is because of `generated by default as identity`. It seems that the console doesnt handle this properly

**Comment by devuxer** [comment]
Note, however, that the table gets altered to `generated always`. That still may not be what Hasura is expecting, though.

**Comment by beerose** [comment]
Can be solved along with: https://github.com/hasura/graphql-engine/pull/4761

**Comment by waterdrop01** [comment]
I confirm, I have `"id" int GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY` on a table, and using the console to insert a new row without specifying the ID results in an error: `Insert failed! invalid input syntax for type integer: ""`

**Comment by rikinsk** [comment]
closed via #4761

**Comment by ryanfabela** [comment]
I am having this issue currently. I am on latest version of Hasura (1.3.2) and 11.8 for my Postgres DB. 
I am trying to insert a new LegalAgreement. I am not inserting into the LegalAgreementId column
Output from Hasura: 
`{
  "errors": [
    {
      "extensions": {
        "internal": {
          "statement": "WITH \"LegalAgreement__mutation_result_alias\" AS (INSERT INTO \"public\".\"LegalAgreement\" ( \"IPAddress\", \"UserProfileId\", \"AcceptedOn\", \"LegalAgreementId\", \"LegalDocumentId\" ) VALUES (($2)::inet, ($1)::integer, ($3)::date, DEFAULT, ($4)::integer), (($6)::inet, ($5)::integer, ($7)::date, DEFAULT, ($8)::integer), (($10)::inet, ($9)::integer, ($11)::date, DEFAULT, ($12)::integer), (($14)::inet, ($13)::integer, ($15)::date, DEFAULT, ($16)::integer), (($18)::inet, ($17)::integer, ($19)::date, DEFAULT, ($20)::integer)  RETURNING * , CASE WHEN 'true' THEN NULL ELSE \"hdb_catalog\".\"check_violation\"('insert check constraint failed')  END ), \"LegalAgreement__all_columns_alias\" AS (SELECT  \"LegalAgreementId\" , \"AcceptedOn\" , \"IPAddress\" , \"UserProfileId\" , \"LegalDocumentId\"  FROM \"LegalAgreement__mutation_result_alias\"      ) SELECT  json_build_object('affected_rows', (SELECT  COUNT(*)  FROM \"LegalAgreement__all_columns_alias\"      ) )        ",
          "prepared": false,
          "error": {
            "exec_status": "FatalError",
            "hint": "Use OVERRIDING SYSTEM VALUE to override.",
            "message": "cannot insert into column \"LegalAgreementId\"",
            "status_code": "428C9",
            "description": "Column \"LegalAgreementId\" is an identity column defined as GENERATED ALWAYS."
          },
          "arguments": [
            {REMOVED_FOR_PRIVACY}
          ]
        },
        "path": "$.selectionSet.insert_Customer.args.objects[0].UserProfile.data.LegalAgreements.data",
        "code": "unexpected"
      },
      "message": "database query error"
    }
  ]
}`

And my table was created using: 
`CREATE TABLE IF NOT EXISTS "LegalAgreement" (
  "LegalAgreementId" int NOT NULL GENERATED ALWAYS AS IDENTITY,
  "AcceptedOn" date NOT NULL,
  "IPAddress" inet NOT NULL,
  "UserProfileId" int NOT NULL,
  "LegalDocumentId" int NOT NULL,
  CONSTRAINT "PK_LegalAgreement" PRIMARY KEY ("LegalAgreementId"),
  CONSTRAINT "FK_539" FOREIGN KEY ("UserProfileId") REFERENCES "UserProfile" ("UserProfileId"),
  CONSTRAINT "FK_556" FOREIGN KEY ("LegalDocumentId") REFERENCES "LegalDocument" ("LegalDocumentId")
);

CREATE UNIQUE INDEX "PK_LegalAgreement" ON "LegalAgreement" ("LegalAgreementId");

CREATE INDEX "fkIdx_539" ON "LegalAgreement" ("UserProfileId");

CREATE INDEX "fkIdx_556" ON "LegalAgreement" ("LegalDocumentId");`
